bundle/direct/dresources/dashboard.go

```diff
@@ -0,0 +1,306 @@
+package dresources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/dashboards"
+)
+
+// Terraform implementation reference: https://github.com/databricks/terraform-provider-databricks/blob/main/dashboards/resource_dashboard.go
+type ResourceDashboard struct {
+	client *databricks.WorkspaceClient
+}
+
+func (*ResourceDashboard) New(client *databricks.WorkspaceClient) *ResourceDashboard {
+	return &ResourceDashboard{client: client}
+}
+
+func (*ResourceDashboard) PrepareState(input *resources.Dashboard) *resources.DashboardConfig {
+	return &input.DashboardConfig
+}
+
+func (r *ResourceDashboard) RemapState(state *resources.DashboardConfig) *resources.DashboardConfig {
+	dashboard := &resources.DashboardConfig{
+		Dashboard: dashboards.Dashboard{
+			DisplayName: state.DisplayName,
+			Etag:        state.Etag,
+			ParentPath:  state.ParentPath,
+			WarehouseId: state.WarehouseId,
+			ForceSendFields: filterFields[dashboards.Dashboard](state.ForceSendFields, []string{
+				"CreateTime",
+				"DashboardId",
+				"LifecycleState",
+				"Path",
+				"UpdateTime",
+				"SerializedDashboard",
+			}...),
+
+			// Clear output only fields. They should not show up on remote diff computation.
+			CreateTime:     "",
+			DashboardId:    "",
+			LifecycleState: dashboards.LifecycleState(""),
+			Path:           "",
+			UpdateTime:     "",
+
+			// Serialized dashboard is ignored for remote diff changes.
+			// They are only relevant for local diff changes.
+			SerializedDashboard: "",
+		},
+
+		EmbedCredentials: state.EmbedCredentials,
+		ForceSendFields: filterFields[resources.DashboardConfig](state.ForceSendFields, []string{
+			"SerializedDashboard",
+		}...),
+
+		// Serialized dashboard is ignored for remote diff changes.
+		// They are only relevant for local diff changes.
+		SerializedDashboard: "",
+	}
+
+	return dashboard
+}
+
+func (r *ResourceDashboard) DoRefresh(ctx context.Context, id string) (*resources.DashboardConfig, error) {
+	wg := sync.WaitGroup{}
+	var dashboard *dashboards.Dashboard
+	var publishedDashboard *dashboards.PublishedDashboard
+	var getErr, getPublishedErr error
+
+	wg.Go(func() {
+		dashboard, getErr = r.client.Lakeview.Get(ctx, dashboards.GetDashboardRequest{
+			DashboardId: id,
+		})
+	})
+	wg.Go(func() {
+		publishedDashboard, getPublishedErr = r.client.Lakeview.GetPublished(ctx, dashboards.GetPublishedDashboardRequest{
```

Comment:
It looks like we make this request only for EmbedCredentials field?

Is that intentional? Can we avoid GetPublished call? (Does EmbedCredentials change?).

Reply:
Thanks, this is good context to add as comments in the code.